tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala

```diff
@@ -2274,4 +2276,17 @@ trait SparkQueryCompareTestSuite extends AnyFunSuite with BeforeAndAfterAll {
   def isCdh330: Boolean = VersionUtils.isCloudera && cmpSparkVersion(3, 3, 0) == 0
 
   def isCdh332: Boolean = VersionUtils.isCloudera && cmpSparkVersion(3, 3, 2) == 0
+
+  // SparkSession => (Boolean, String)
+  def ignoreAnsi(issue: String)(spark: SparkSession): (Boolean, String) = {
+    (!SQLConf.get.ansiEnabled, s"ANSI mode is not supported in this test: ${issue}")
+  }
+
+  def skipIfAnsiEnabled(issue: String): Unit = {
+    // Initialize SparkSessionHolder.sparkSession to ensure environment variables from SPARK_CONF
+    // are properly loaded before checking ANSI settings
+    SparkSessionHolder.sparkSession
+    val ansiEnabled = SQLConf.get.getConf(SQLConf.ANSI_ENABLED)
+    assume(!ansiEnabled, s"ANSI mode is not supported in this test: ${issue}")
+  }
```

Comment:
This method is used outside an active session. However, everywhere it is used there is an active session just a couple lines below. You can redefine it here in terms of ignoreAnsi

```suggestion
  def skipIfAnsiEnabled(issue: String): Unit = {
    val (isNonAnsi, skipReason) = ignoreAnsi(issue)(null)
    assume(isNonAnsi, skipReason)
  }
```

Reply:
Thanks @gerashegalov ! We would need to initialize sparksession if it not initialized at all when this function(ignoreAnsi) is called. 
For example if I am running individual test and if the SparkSession is not initialized then `spark.sql.ansi.enabled` would return `true` because the code to get the configs from  SPARK_CONF is not run.
 
With the above suggested code, when I run OrcFilterSuite it would skip the first test (for `test-ansi-off`) mode. 
```
INFO] --- scalatest:2.0.2:test (test-ansi-off) @ rapids-4-spark-tests_2.13 ---
Discovery starting.
Discovery completed in 225 milliseconds.
Run starting. Expected test count is: 9
OrcFilterSuite:
- Support for pushing down filters for boolean types gpu write gpu read !!! CANCELED !!!
  isNonAnsi was false ANSI mode is not supported in this test: https://github.com/NVIDIA/spark-rapids/issues/5114 (SparkQueryCompareTestSuite.scala:2296)
- Support for pushing down filters for boolean types gpu write cpu read
- Support for pushing down filters for boolean types cpu write gpu read

```
So I think we need to include  `SparkSessionHolder.sparkSession`. So `skipIfAnsiEnabled` can be:

```
  def skipIfAnsiEnabled(issue: String): Unit = {
    SparkSessionHolder.sparkSession
    val (isNonAnsi, skipReason) = ignoreAnsi(issue)(null)
    assume(isNonAnsi, skipReason)
  }
```
Please let me know your thoughts on this.

Reply:
My idea was to move skipIfAnsiEnabled inside with*Session so we don't have to manually touch the lazy SparkSessionHolder.sparkSession. 

Your proposal also LGTM, at least we are not duplicating the condition logic for `assume`